src/QsCompiler/LanguageServer/LanguageServer.cs

```diff
@@ -123,9 +124,44 @@ public void Dispose()
         internal Task NotifyClientAsync(string method, object args) =>
             this.rpc.NotifyWithParameterObjectAsync(method, args);  // no need to wait for completion
 
+        internal Task<T> InvokeAsync<T>(string method, object args) =>
+            this.rpc.InvokeWithParameterObjectAsync<T>(method, args);
+
         internal Task PublishDiagnosticsAsync(PublishDiagnosticParams diagnostics) =>
             this.NotifyClientAsync(Methods.TextDocumentPublishDiagnosticsName, diagnostics);
 
+        internal async void CheckDotNetSdkVersion()
+        {
+            var sdkVersions = DotNetSdkHelper.GetSdkVersions();
+            if (sdkVersions == null)
+            {
+                this.LogToWindow($"Unable to detect .NET SDK versions", MessageType.Error);
+            }
+            else
+            {
+                if (!sdkVersions.HasDotNet31)
+                {
+                    const string dotnet31Url = "https://dotnet.microsoft.com/download/dotnet-core/3.1";
+                    this.LogToWindow($".NET SDK 3.1 not found. Quantum Development Kit Extension requires .NET Core SDK 3.1 to work properly ({dotnet31Url}).", MessageType.Error);
+                    var downloadAction = new MessageActionItem { Title = "Download" };
+                    var cancelAction = new MessageActionItem { Title = "No, thanks" };
+                    var selectedAction = await this.ShowDialogInWindowAsync(
+                        $"Quantum Development Kit Extension requires .NET Core SDK 3.1 to work properly. Please install .NET SDK 3.1.",
```

Comment:
I think we should consistently write ".NET Core SDK 3.1" instead of ".NET SDK 3.1".

Reply:
Done